tools/mason/MasonHelp.chpl

```diff
@@ -105,6 +105,7 @@ proc masonBuildHelp() {
 proc masonNewHelp() {
   writeln('Usage:');
   writeln('    mason new [options] <project name>');
+  writeln('    mason new --interactive');
```

Comment:
Depending on our answer to https://github.com/chapel-lang/chapel/pull/15748#pullrequestreview-424805913 - we may want to update the help output.

Reply:
@ben-albrecht I think we should give an error. ( just to keep it clean, and since package name is the first question anyway so no point in skipping it, we'd still require inputs for version and chplVersion.  )

Reply:
On more thought, I am actually leaning towards supporting it and skipping the package step as a result (though still doing the error checking).

This will reduce the chance of the user stumbling over an error, and will also allow us to revert the documentation back into the pre-forked version that your originally had:

```
mason new [options] <project name>');
```

where `--interactive` is just another argument.

I see the merit in your argument of keeping the implementation cleaner. I think the resulting cleaner user interface will be worth it though.

Reply:
Should I proceed with the error message here ?

Reply:
> On more thought, I am actually leaning towards supporting it and skipping the package step as a result (though still doing the error checking and throwing an error if the package name has any issues like we normally do for non-interactive mason new).

Thanks for clarifying. I was advocating for (c) from https://github.com/chapel-lang/chapel/pull/15748#pullrequestreview-424805913

Do you have any opposition to that approach?

Reply:
Okay, should we do the same with `mason init` ?

Reply:
I think so, since the directory name == package name by default.

Reply:
> Do you have any opposition to that approach?

I don't really oppose it but I feel if an user is using interactive, they must be new to Chapel development. So I don't really think they'd actually go for using a name within the command.
That was why I suggested (a). 
If we consider adding name within the command, there might still be a case when the user has actually added a name there by mistake, and they might want to change it. Considering that, I'd also say (b) is a good option. What do you think ?

Reply:
That's a good argument. I agree and think (b) is the most ideal.

Reply:
I also had a thought while developing this feature about dropping the `--interactive` flag. I think it would be a good idea to start interactive session if an user simply types `mason new`(outside directory) or `mason init` (inside directory). As an user, I think its more effort to type the `interactive` option.

Reply:
I think using this method will separate this feature completely from the previous versions of mason init & mason new. If an user uses a name or option, commands will work as they should.

Reply:
Oh, I really like that idea.

Just to be clear, `mason new` would not be impacted, but `mason init` behavior changes, right? Previously `mason init` just used the default values, and now it would start an interactive session. Not a big change, but a change nevertheless.

Curious what @Spartee and/or @krishnadey30 thinks.

Reply:
>  Previously mason init just used the default values, and now it would start an interactive session.

For `mason init` we could simply check if the directory where it's called is empty or not. If its empty, we start an interactive session else it behaves normally.

Reply:
I support this. This thought actually crossed my mind when I was playing around with it. 

@ben-albrecht Why should ``mason new`` not also default to interactive?

Reply:
I am also in favor of this.

Reply:
@ankingcodes  - Sounds like you may proceed on this feature for `mason new`. Great idea!

For `mason init`, I am still uncertain on some details. It seems odd/unpredictable for the command to have totally different behaviors based on what files are present rather than arguments, e.g.

```sh
> mkdir Foo
> mason init
# interactive session
^C
> touch Mason.toml
> mason init
# automatically chooses defaults in non-interactive session
```

A few ways we could make this behavior more predictable are:

1. `mason init` is always interactive no matter what.
2. `mason init` supports the `--interactive` flag and is non-interactive by default (how the PR currently is)

Do you have any thoughts?

Reply:
@ben-albrecht I'd go with (1) since the idea is to get rid of the flag. 
> mason init is always interactive no matter what.

A small problem with using interactive if Mason.toml is present is that, we will require it then to overwrite it with the values given by an user. I know this has been discussed and is potential future feature, but I also think it might be irritating for an user to pass through the entire interactive session just to initialise their project. So, that was why I think the method I suggested above is a better way to tackle this. 
I'd like to improve it as - if a `Mason.toml && src/module.chpl` file is present, simply initialise a project, otherwise, run interactive. The reason to consider those two files is because, they're the only ones affected by this feature. 

Another option that comes to mind is introducing a `--no-interactive` flag to override the interactive `mason init`. But I'm not a big fan of this option. 

**The clash between interactive and init is that, init doesnt support overwriting, while interactive supports overwriting.** Would you be in support of keeping this feature only for `mason new` ? If `mason new` is used outside a directory, it'd work normally. If `mason new` is used inside an empty directory, interactive works as it does for `mason init`. If it finds `Mason.toml` or `src/module.chpl` it prints an error informing users that mason new should be used only in an empty directory. 

I'd be interested to know your opinions on these options.

Reply:
> A small problem with using interactive if Mason.toml is present is that, we will require it then to overwrite it with the values given by an user. 

I think it makes sense for `mason init` to not overwrite any existing files/`Mason.toml` entries for the non-interactive mode where we are using the defaults with no user input. However, for interactive mode, it seems reasonable to overwrite the existing `Mason.toml` values, because the user is able to give input on what those overwritten values should store.

> I know this has been discussed and is potential future feature, but I also think it might be irritating for an user to pass through the entire interactive session just to initialise their project.

I think coming back to the earlier idea of using any pre-existing `Mason.toml` fields as a default value for the interactive session would alleviate this issue. That said, I understand this is a big change which we may not want to tackle in this PR.

> Another option that comes to mind is introducing a --no-interactive flag to override the interactive mason init. But I'm not a big fan of this option.

I feel similarly about this.

> Would you be in support of keeping this feature only for mason new ?

I think this would be a reasonable path forward for keeping the scope of this PR manageable. This would give us more time to think about how we want `mason init` to behave too.

So to recap, the proposal is:

- `mason new` with no _positional_ arguments (flags are OK) uses the new interactive mode
- `mason init` remains unchanged, but we will plan to support interactive mode in the future once we reach design consensus in a separate issue.

Does that sound good to you?

Reply:
Sounds good ! I'll go ahead and complete this PR now. 👍